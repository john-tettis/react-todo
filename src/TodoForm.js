import React, {useState} from 'react'
import { Button, Form, FormGroup, Label, Input} from 'reactstrap';
import 'bootstrap/dist/css/bootstrap.min.css';


let TodoForm = ({createTodo})=>{
    let [value, setValue] = useState('')
    let [valid,setValid] = useState(false)
    let [touched, setTouched] = useState(false)
    let updateVal=(e)=>{
        setValue(e.target.value)
        if(e.target.value===''){
            setValid(false);
        }
        else{
            setValid(true);
        }
        setTouched(true)
    }
    let resetValue= ()=>{setValue('')}
    let handleSubmit =(e)=>{
        e.preventDefault()
        if(valid){
            createTodo(value)
            resetValue()
            setValid(false)
            setTouched(false)
        }
        else{
            setTouched(true)
        }
    }
        

    return(
        <Form className='mt-4' onSubmit={(e)=>{handleSubmit(e)}}>
        <FormGroup>
            <Label className='mb-2'for="todo">Create Todo</Label>
            <Input valid={valid} type="text" onChange={updateVal}name="todo" id="todo" value={value} placeholder="What to do..." />
            {!valid && touched &&
            <small className='text-danger'>Todo cannot be empty!</small>
            }   
        </FormGroup>
        <Button color='primary'className='mt-2'>Create</Button>
    </Form>
        
    )
}

export default TodoForm;