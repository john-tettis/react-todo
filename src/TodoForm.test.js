import { render, screen } from '@testing-library/react';
import TodoForm from './TodoForm';

test('renders ok', () => {
  render(<TodoForm />);
});

test('snapshot is consistent', () => {
  let {asFragment}=render(<TodoForm />);
  expect(asFragment()).toMatchSnapshot()
});