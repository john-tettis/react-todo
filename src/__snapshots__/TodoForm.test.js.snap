// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot is consistent 1`] = `
<DocumentFragment>
  <form
    class="mt-4"
  >
    <div
      class="form-group"
    >
      <label
        class="mb-2"
        for="todo"
      >
        Create Todo
      </label>
      <input
        class="form-control"
        id="todo"
        name="todo"
        placeholder="What to do..."
        type="text"
        value=""
      />
    </div>
    <button
      class="mt-2 btn btn-primary"
    >
      Create
    </button>
  </form>
</DocumentFragment>
`;
